src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -334,17 +334,27 @@ protected virtual void Initialize()
                     // If there's a parsing error, set up the logger and write out the previous messages so that they're
                     // discoverable in Application Insights. There's no filter, but that's okay since we cannot parse host.json to
                     // determine how to build the filtler.
-                    ConfigureLoggerFactory(ScriptConfig, FunctionTraceWriterFactory, _settingsManager, () => FileLoggingEnabled);
```

Comment:
Have you considered making LoggerFactory a Lazy with this as the default factory function, or something along those lines so that we don't need to call ConfigureDefaultLoggerFactory() in multiple locations?